src/Sets/Line.jl

```diff
@@ -0,0 +1,455 @@
+ambient dimension of `L`.
+"""
+function constraints_list(L::Line{N, VN}) where {N, VN}
+    p = L.p
+    d = length(p)
+    n = reshape(L.n, 1, d)
+    K = nullspace(n)
+    m = size(K, 2)
+    @assert m == d - 1 "expected $(d - 1) normal half-spaces, got $m"
+
+    out = Vector{HalfSpace{N, VN}}(undef, 2m)
+    idx = 1
+    @inbounds for j in 1:m
+        Kj = K[:, j]
+        b = dot(Kj, p)
+        out[idx] = HalfSpace(Kj, b)
+        out[idx+1] = HalfSpace(-Kj, -b)
+        idx += 2
+    end
+    return out
+end
+
+# --- LazySet interface functions ---
+
+
+"""
+    dim(L::Line)
+
+Return the ambient dimension of a line.
+
+### Input
+
+- `L` -- line
+
+### Output
+
+The ambient dimension of the line.
+"""
+dim(L::Line) = length(L.p)
+
+"""
+    ρ(d::AbstractVector, L::Line)
+
+Return the support function of a line in a given direction.
+
+### Input
+
+- `d` -- direction
+- `L` -- line
+
+### Output
+
+The support function in the given direction.
+"""
+function ρ(d::AbstractVector, L::Line)
+    if isapproxzero(d, L.n)
+        return dot(d, L.p)
+    else
+        return Inf
+    end
+end
+
+"""
+    σ(d::AbstractVector{N}, L::Line{N}) where {N<:Real}
+
+Return the support vector of a line in a given direction.
+
+### Input
+
+- `d` -- direction
+- `L` -- line
+
+### Output
+
+The support vector in the given direction.
+"""
+function σ(d::AbstractVector, L::Line)
+    if isapproxzero(d, L.n)
+        return L.p
+    else
+        error("the support vector is undefined because the line is unbounded")
+    end
+end
+
+"""
+    isbounded(L::Line)
+
+Determine whether a line is bounded.
+
+### Input
+
+- `L` -- line
+
+### Output
+
+`false`.
+"""
+isbounded(::Line) = false
+
+"""
+    isuniversal(L::Line; [witness::Bool]=false)
+
+Check whether a line is universal.
+
+### Input
+
+- `P`       -- line
+- `witness` -- (optional, default: `false`) compute a witness if activated
+
+### Output
+
+* If `witness` option is deactivated: `false`
+* If `witness` option is activated: `(false, v)` where ``v ∉ P``
+"""
+isuniversal(L::Line; witness::Bool=false) = isuniversal(L, Val(witness))
+
+isuniversal(L::Line, ::Val{false}) = false
+
+"""
+    an_element(L::Line)
+
+Return some element of a line.
+
+### Input
+
+- `L` -- line
+
+### Output
+
+An element on the line.
+"""
+an_element(L::Line) = L.p
+
+"""
+    ∈(x::AbstractVector, L::Line)
+
+Check whether a given point is contained in a line.
+
+### Input
+
+- `x` -- point/vector
+- `L` -- line
+
+### Output
+
+`true` iff `x ∈ L`.
+
+### Algorithm
+
+The point ``x`` belongs to the line ``L : p + λ⋅n`` if and only if
+``x - p`` is proportional to the direction ``n``.
+"""
+function ∈(x::AbstractVector, L::Line)
+    @assert length(x) == dim(L) "expected the point and the line to have the same dimension, " *
+                                "but they are $(length(x)) and $(dim(L)) respectively"
+    _isapprox(x, L.p) && return true
+    return first(samedir(x - L.p, L.n))
```

Comment:
i think that's included, because if there exists λ such that `x - L.p = λ * (-L.n)`, then for `μ := -λ`,  `x - L.p = μ * L.n`.

Reply:
```julia
julia> using LazySets: samedir

julia> x = [1.]
1-element Array{Float64,1}:
 1.0

julia> p = [2.]
1-element Array{Float64,1}:
 2.0

julia> n = [3.]
1-element Array{Float64,1}:
 3.0

julia> samedir(x - p, n)
(false, 0)

julia> samedir(x - p, -n)
(true, 0.3333333333333333)
```

Reply:
ok :) i didn't expect `samedir`  to only use positive constants. we could extend `samedir` for this purpose (optionally at least)

Reply:
Okay, but we should also keep the old behavior or at least check where `samedir` is used to not break something else. A new function to avoid confusion is also fine.